sriovnet_switchdev.go

```diff
@@ -226,6 +226,36 @@ func findNetdevWithPortNameCriteria(criteria func(string) bool) (string, error)
 	return "", fmt.Errorf("no representor matched criteria")
 }
 
+// GetPortIndexFromRepresentor finds the index of a representor from its network device name.
+// Supports VF and SF.
+func GetPortIndexFromRepresentor(repNetDev string) (int, error) {
+	flavor, err := GetRepresentorPortFlavour(repNetDev)
+	if err != nil {
+		return 0, err
+	}
+
+	if flavor != PORT_FLAVOUR_PCI_VF && flavor != PORT_FLAVOUR_PCI_SF {
+		return 0, fmt.Errorf("device %s is not an indexed representor port", repNetDev)
+	}
+
+	physPortName, err := getNetDevPhysPortName(repNetDev)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get device %s physical port name: %v", repNetDev, err)
+	}
+
+	typeToRegex := map[PortFlavour]*regexp.Regexp{
+		PORT_FLAVOUR_PCI_VF: vfPortRepRegex,
+		PORT_FLAVOUR_PCI_SF: sfPortRepRegex,
```

Comment:
Thinking on this more, we may get the same port index for different types of port flavours

say we would want later on to extend it to PORT_FLAVOUR_PCI_PF and PORT_FLAVOUR_PCI_PHYSICAL

in case of VF we can have phys_port_name as `pf0vf0`
in case of SF we can have phys_port_name as `pf0sf0`
in case of uplink rep we would have physical_port_name as `p0`
in case of PF rep we would have physical_port_name as `pf0hpf`

in all the above would their "port index" be zero?
i assume answer is yes and caller will call `GetRepresentorPortFlavor` if needed to differentiate.

maybe add some additional info in function comments saying that the same index may be returned for representors of different flavors and `GetRepresentorPortFlavor` should be used to differentiate between them.

Reply:
Yes, for each of those the returned ID will be 0.

I'll add a comment.